WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -553,3 +565,62 @@ private extension MainTabBarController {
         viewModel.startObservingOrdersCount()
     }
 }
+
+// MARK: - Background Product Image Upload Status Updates
+
+private extension MainTabBarController {
+    func observeProductImageUploadStatusUpdates() {
+        guard featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) else {
+            return
+        }
+        productImageUploadStatusUpdatesSubscription = productImageUploader.statusUpdates.sink { [weak self] update in
+            self?.handleBackgroundImageUploadUpdate(update)
+        }
+    }
+
+    func handleBackgroundImageUploadUpdate(_ update: ProductImageUploadUpdate) {
+        if update.error != nil {
+            let notice = Notice(title: Localization.imageUploadFailureNoticeTitle,
+                                subtitle: nil,
+                                message: nil,
+                                feedbackType: .error,
+                                notificationInfo: nil,
+                                actionTitle: Localization.imageUploadFailureNoticeActionTitle,
+                                actionHandler: { [weak self] in
+                guard let self = self else { return }
+                Task { @MainActor in
+                    await self.showProductDetails(update: update)
+                }
+            })
+            noticePresenter.enqueue(notice: notice)
+        }
+    }
+
+    func showProductDetails(update: ProductImageUploadUpdate) async {
+        // Switches to the correct store first if needed.
+        let switchStoreUseCase = SwitchStoreUseCase(stores: stores)
+        let siteChanged = await switchStoreUseCase.switchStore(with: update.siteID)
+        if siteChanged {
+            let presenter = SwitchStoreNoticePresenter(siteID: update.siteID,
+                                                       noticePresenter: self.noticePresenter)
+            presenter.presentStoreSwitchedNoticeWhenSiteIsAvailable(configuration: .switchingStores)
+        }
```

Comment:
Nice work covering the switching store scenario. 🙇 

Would it make sense to move the switch store logic to a separate function?

Reply:
I feel like these lines of code are pretty straightforward and creating a separate function might be easier to read 🤔 lemme know if you think otherwise!